app/dashboard/views/fido_setup.py

```diff
@@ -0,0 +1,95 @@
+import uuid
+import json
+import secrets
+import webauthn
+from app.config import RP_ID
+from urllib.parse import urlparse
+
+from flask import render_template, flash, redirect, url_for, session
+from flask_login import login_required, current_user
+from flask_wtf import FlaskForm
+from wtforms import HiddenField, validators
+
+from app.dashboard.base import dashboard_bp
+from app.extensions import db
+from app.log import LOG
+
+
+class FidoTokenForm(FlaskForm):
+    sk_assertion = HiddenField("sk_assertion", validators=[validators.DataRequired()])
+
+
+@dashboard_bp.route("/fido_setup", methods=["GET", "POST"])
+@login_required
+def fido_setup():
+    if current_user.fido_uuid is not None:
+        flash("You have already registered your security key", "warning")
+        return redirect(url_for("dashboard.index"))
+
+    fido_token_form = FidoTokenForm()
+
+    # Handling POST requests
+    if fido_token_form.validate_on_submit():
+        try:
+            sk_assertion = json.loads(fido_token_form.sk_assertion.data)
+        except Exception as e:
+            flash("Key registration failed. Error: Invalid Payload", "warning")
+            return redirect(url_for("dashboard.index"))
+
+        fido_uuid = session["fido_uuid"]
+        challenge = session["fido_challenge"]
+
+        fido_reg_response = webauthn.WebAuthnRegistrationResponse(
+            RP_ID,
+            SITE_URL,
```

Comment:
`SITE_URL` is unresolved here

Reply:
e4895b52a00ca2bb9810858d33199392652d0d36